rkt/image_list.go

```diff
@@ -194,52 +207,67 @@ func runImages(cmd *cobra.Command, args []string) int {
 			continue
 		}
 		version, ok := im.Labels.Get("version")
-		var fieldValues []string
-		for _, f := range flagImagesFields.Options {
-			fieldValue := ""
-			switch f {
-			case l(id):
-				hashKey := aciInfo.BlobKey
-				if !flagFullOutput {
-					// The short hash form is [HASH_ALGO]-[FIRST 12 CHAR]
-					// For example, sha512-123456789012
-					pos := strings.Index(hashKey, "-")
-					trimLength := pos + 13
-					if pos > 0 && trimLength < len(hashKey) {
-						hashKey = hashKey[:trimLength]
-					}
-				}
-				fieldValue = hashKey
-			case l(name):
-				fieldValue = aciInfo.Name
-				if ok {
-					fieldValue = fmt.Sprintf("%s:%s", fieldValue, version)
-				}
-			case l(importTime):
-				if flagFullOutput {
-					fieldValue = aciInfo.ImportTime.Format(defaultTimeLayout)
-				} else {
-					fieldValue = humanize.Time(aciInfo.ImportTime)
-				}
-			case l(lastUsed):
-				if flagFullOutput {
-					fieldValue = aciInfo.LastUsed.Format(defaultTimeLayout)
-				} else {
-					fieldValue = humanize.Time(aciInfo.LastUsed)
-				}
-			case l(size):
-				totalSize := aciInfo.Size + aciInfo.TreeStoreSize
-				if flagFullOutput {
-					fieldValue = fmt.Sprintf("%d", totalSize)
-				} else {
-					fieldValue = humanize.IBytes(uint64(totalSize))
-				}
-			case l(latest):
-				fieldValue = fmt.Sprintf("%t", aciInfo.Latest)
+
+		imageID := aciInfo.BlobKey
+		imageName := aciInfo.Name
+		if ok {
+			imageName = fmt.Sprintf("%s:%s", imageName, version)
+		}
+
+		imageImportTime := humanize.Time(aciInfo.ImportTime)
+		imageLastUsed := humanize.Time(aciInfo.LastUsed)
+		totalSize := aciInfo.Size + aciInfo.TreeStoreSize
+		imageSize := humanize.IBytes(uint64(totalSize))
+
+		if flagFullOutput {
+			imageImportTime = aciInfo.ImportTime.Format(defaultTimeLayout)
+			imageLastUsed = aciInfo.LastUsed.Format(defaultTimeLayout)
+			imageSize = fmt.Sprintf("%d", totalSize)
+		} else {
+			// The short hash form is [HASH_ALGO]-[FIRST 12 CHAR]
+			// For example, sha512-123456789012
+			pos := strings.Index(imageID, "-")
+			trimLength := pos + 13
+			if pos > 0 && trimLength < len(imageID) {
+				imageID = imageID[:trimLength]
+			}
```

Comment:
Also for the JSON case, I would not trim the imageID, this also retains symmetry with `rkt list`, which also ignores `rkt list --full --format=json`.